### ISSUE TITLE
Support syntax highlighting for `impl const Trait for Ty`

### ISSUE THREAD

**Issue by CDirkx** [intellij-rust/intellij-rust#5059]
I dont know what the plugins policy is on supporting experimental Rust features, so if this is out of scope let me know.

Experimental implementation has started of const traits: [rust-lang#67792](https://github.com/rust-lang/rust/issues/67792), gated by `const_trait_impl`. The [RFC](https://github.com/rust-lang/rfcs/pull/2632) this implementation is based on has **not** been approved yet, and this implementation is done to evaluate the design. There is thus still a long way to go before any of this stabilizes.

However, I believe that with a small change to the plugins syntax highlighting, we can support this feature partially, reducing friction for users to test the feature and providing the Rust team with valuable feedback: allow the keyword `const` to follow `impl`.

```rust
impl const Trait for Ty { }
```

Currently this disrupts most of the syntax highlighting and code insights (go to definition, show usages, implement methods, etc.) of code that comes after.

`impl`: full syntax highlighting and code insights
![impl](https://user-images.githubusercontent.com/11176562/76151834-723c6b80-60b9-11ea-8391-64e3babcccfa.png)

`impl const`: limited syntax highlighting and no code insight
![impl_const](https://user-images.githubusercontent.com/11176562/76151837-72d50200-60b9-11ea-9f1b-96a64494f3e2.png)
compiles fine on the latest nightly (with `const_trait_impl` enabled)
